neuralprophet/time_dataset.py

```diff
@@ -377,13 +384,21 @@ def _stride_lagged_features(df_col_name, feature_dims):
                 events["multiplicative"] = multiplicative_events
 
         inputs["events"] = events
-
     if predict_mode:
         targets = np.empty_like(time)
         targets = np.nan_to_num(targets)
     else:
         targets = _stride_time_features_for_forecasts(df["y_scaled"].values)
 
+    if forecast_period is not None:
```

Comment:
I wonder if we could make this a dedicated method to `TimeDataset`?
(called after `init_after_tabularized`)
Or, we could add it to `init_after_tabularized` to avoid another loop. 

Both methods would keep the already massive tabularization function leaner.

What do you think @judussoari @noxan ?

Reply:
Great suggestion, did that!